server/searchHandler.go

```diff
@@ -0,0 +1,74 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-github/github"
+	pb "github.com/kruckenb/githubSearch/proto"
+	"golang.org/x/oauth2"
+	"log"
+	"net/http"
+)
+
+type server struct {
+	oauthToken *string
+
+	pb.UnimplementedGithubSearchServiceServer
+}
+
+func (s *server) Search(ctx context.Context, req *pb.SearchRequest) (*pb.SearchResponse, error) {
+	log.Printf("Request %v", req)
+	results, err := searchGithub(ctx, req, githubClient(ctx, s.oauthToken))
+	log.Printf("Returning results: %v", results)
+	if err != nil {
+		return nil, err
+	} else {
+		return &pb.SearchResponse{ Results: results }, nil
+	}
+}
+
+func githubClient(ctx context.Context, oauthToken *string) *github.Client {
+	if oauthToken != nil && *oauthToken != "" {
+		ts := oauth2.StaticTokenSource(
+			&oauth2.Token{AccessToken: *oauthToken},
+		)
+		return github.NewClient(oauth2.NewClient(ctx, ts))
+	} else {
+		return github.NewClient(nil)
+	}
+}
+
+func searchGithub (ctx context.Context, req *pb.SearchRequest, client *github.Client) ([]*pb.Result, error) {
```

Comment:
You may be seeing my Java years leaking through on naming - though I did repress the urge to name anything as a BeanFactory. I tried to get close to "the Go way" with names, but maybe not all the way.